planetary-social/nos
# Content flag category selection

## Issues covered
Part 2 of [#1498](https://github.com/planetary-social/nos/issues/1498)
Part 1 of [#1489](https://github.com/planetary-social/nos/issues/1489)

## Description
This PR starts work on the new content flag flow. It includes the following changes:
- A new `NosRadioButton` component.
- A new `FlagOptionPicker` component.
- A new `ContentFlagView` component.
- A new Moderation folder under the views folder.
- Updates `ReportMenuModifier` to show either the old menu or new menu based on the `newModerationFlow` feature flag boolean.

## How to test
1. Build the app
2. Click the side menu
3. Click the Settings on the side menu
4. Scroll down to turn on the toggle for new moderation flow
5. Go back to the Feed screen by clicking the feed tab at the bottom of the screen
6. Click on the 3 dots on one of the notes.
7. Select `Flag this content` option. 
8. Please confirm that the new menu for the content flag category shows up.

## Screenshots/Video
Before:

https://github.com/user-attachments/assets/ab1e3985-9dd1-4ff3-ac72-a4c307e07a09

After:

https://github.com/user-attachments/assets/93be9a46-6afa-4485-b867-4f8eba1509ae
